ribs/emitters/rankers.py

```diff
@@ -0,0 +1,368 @@
+"""Package with rankers for use across emitters.
+
+The rankers implemented in this file is intended to be used with emitters.
+Specifically, a ranker object should be initialized or passed in the emitters.
+The ``Ranker`` object will define the :meth:`rank` method which returns the
+result of an descending argsort of the solutions. It will also define a
+:meth:`reset` method which resets the internal states of the object.
+
+.. autosummary::
+    :toctree:
+
+    ribs.emitters.rankers.RandomDirectionRanker
+    ribs.emitters.rankers.TwoStageRandomDirectionRanker
+    ribs.emitters.rankers.ObjectiveRanker
+    ribs.emitters.rankers.TwoStageObjectiveRanker
+    ribs.emitters.rankers.ImprovementRanker
+    ribs.emitters.rankers.TwoStageImprovementRanker
+    ribs.emitters.rankers.RankerBase
+"""
+
+from abc import ABC, abstractmethod
+import numpy as np
+
+from ribs._docstrings import DocstringComponents, _core_docs
+
+__all__ = [
+    "RandomDirectionRanker",
+    "TwoStageRandomDirectionRanker",
+    "ObjectiveRanker",
+    "TwoStageObjectiveRanker",
+    "ImprovementRanker",
+    "TwoStageImprovementRanker",
+    "RankerBase",
+]
+
+# Define common docstrings
+_args = _core_docs["args"]
+_returns = DocstringComponents(
+    dict(index_batch="""
```

Comment:
Let's only use `index_batch` when referring to a batch of indices from the archive.